packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -689,6 +703,71 @@ export class Cluster extends ClusterBase {
     }
   }
 
+  /**
+   * Enables automatic cluster rebooting when changes to the cluster's parameter group require a restart to apply.
+   */
+  public enableRebootForParameterChanges(): void {
+    if (!this.rebootForParameterChangesEnabled) {
+      this.rebootForParameterChangesEnabled = true;
```

Comment:
The check on line 607 is all we need for this, no?

Reply:
I believe we need it to make sure we don't make a [duplicate Custom Resource](https://github.com/aws/aws-cdk/pull/22063/files/1c6918595a454ad46b9918b3848be2792e1eb060#diff-1ae118454a3ef75fce1f5c54ea531575d0b77d43bb88b914898ccd93aac3c15aR734-R737). I could create class variables to make the function idempotent, but thought that may be overkill

Reply:
making the function idempotent is definitely overkill, and guarding against these potential bugs (an error would be thrown complaining about duplicate construct IDs)  is definitely the right call in general. In this case though I don't see how the function could be called twice, as it's only called in the constructor and it's internal-only.

Reply:
I originally thought about making it constructor only, but It's currently not [(example)](https://github.com/aws/aws-cdk/pull/22063/files/80848a95b82da7ac917b6766b5bd4a9b902d0ec1#diff-adc7742a496eece1305edc13496af7b2cfa60b7bf95f14a1852ffa62d438af08R665-R685). I figured that exposing the method would be friendlier for situationally enabling the feature under specific conditions (ex. beta/prod stages,  using the addToParameterGroup method and then using this method).

Reply:
hmmm okay. I don't love that we have this prop and this internal member, but it is a better user experience to be able to use a convenience method to set it, so we can leave it. 

Please change the format of this to a guard clause though, eg:

```
    if (this.rebootForParameterChangesEnabled) {
      return;
    }
    this.rebootForParameterChangesEnabled = true;
```